code/game/machinery/ds13/stasis_station.dm

```diff
@@ -0,0 +1,61 @@
+/obj/machinery/stasis_station
+	name = "stasis recharge station"
+	desc = "This station is used to recharge stasis module."
+	icon = 'icons/obj/machines/ds13/stasis.dmi'
+	icon_state = "stasis_station"
+	anchored = 1
+	use_power = 0
+
+	var/busy = FALSE
+	var/recharging = FALSE
+
+/obj/machinery/stasis_station/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/machinery/stasis_station/update_icon()
+	if(busy)
+		icon_state = "stasis_station_discharging"
+	else if(recharging)
+		icon_state = "stasis_recharging"
+	else
+		icon_state = "stasis_station"
+
+/obj/machinery/stasis_station/attack_hand(var/mob/user)
+	if(busy)
+		to_chat(user, SPAN_NOTICE("Someone is already using [src]"))
+	else if(recharging)
+		to_chat(user, SPAN_NOTICE("[src] is recharging!"))
+		playsound(loc, 'sound/machines/buzz-two.ogg', VOLUME_MID, 0)
+	else
+		if(istype(user.back, /obj/item/weapon/rig))
```

Comment:
don't check the back, look at the wearing_rig var instead.
It will only ever contain a rig or null so no typechecking needed, and it accomodates the possibility of rigs worn in other slots